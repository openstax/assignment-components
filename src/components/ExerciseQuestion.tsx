import styled from "styled-components";
import { ID, Question as QuestionType, Task } from "../types";
import { Content } from "./Content";
import { Question } from './Question';
import { StepCardFooter } from "./StepCardFooter";

export interface ExerciseQuestionProps {
  task?: Task;
  question: QuestionType;
  answer_id_order?: ID[];
  questionNumber: number;
  choicesEnabled: boolean;
  hasMultipleAttempts: boolean;
  onAnswerChange: () => void;
  onAnswerSave: () => void;
  onNextStep: () => void;
  feedback_html: string;
  correct_answer_feedback_html: string;
  is_completed: boolean;
  correct_answer_id: ID;
  incorrectAnswerId: ID;
  multiPartGroup: any;
  answerId: ID;
  attempts_remaining: number;
  published_comments: string;
  detailedSolution: string;
  canAnswer: boolean;
  needsSaved: boolean;
  canUpdateCurrentStep: boolean;
  attempt_number: number;
  apiIsPending: boolean;
  displaySolution: boolean;
  available_points: string;
  exercise_uid: string;
}

// @TODO
const Button = styled.button`
`;
const AsyncButton = (props: any) => {
  return <Button>{props.children}</Button>;
}

const AttemptsRemaining = ({ count }: { count: number }) => {
  return (
    <div>{count} attempt{count === 1 ? '' : 's'} left</div>
  );
}

const PublishedComments = ({ published_comments }: { published_comments: string }) => {
  if (!published_comments) { return null; }

  return (
    <div>
      <strong>Feedback:</strong> {published_comments}
    </div>
  );
}

const SaveButton = (props: {
  disabled: boolean, isWaiting: boolean, onAnswerSave: ExerciseQuestionProps['onAnswerSave'],
  attempt_number: number
}) => (
  <AsyncButton
    waitingText="Savingâ€¦"
    {...props}
    data-test-id="submit-answer-btn"
  >
    {props.attempt_number == 0 ? 'Submit' : 'Re-submit'}
  </AsyncButton>
);

const NextButton = (props: {
  canUpdateCurrentStep: boolean,
  onNextStep: ExerciseQuestionProps['onNextStep']
}) => {
  return (
    <Button onClick={props.onNextStep} data-test-id="continue-btn">
      {props.canUpdateCurrentStep ? 'Continue' : 'Next'}
    </Button>
  );
}

export const ExerciseQuestion = (props: ExerciseQuestionProps) => {
  const {
    question, task, answer_id_order, onAnswerChange, feedback_html, correct_answer_feedback_html,
    is_completed, correct_answer_id, incorrectAnswerId, choicesEnabled, questionNumber,
    answerId, hasMultipleAttempts, attempts_remaining, published_comments, detailedSolution,
    canAnswer, needsSaved, attempt_number, apiIsPending, onAnswerSave, onNextStep, canUpdateCurrentStep,
    displaySolution, available_points
  } = props;

  return (
    <div data-test-id="student-exercise-question">
      <Question
        task={task}
        question={question}
        answerIdOrder={answer_id_order}
        choicesEnabled={choicesEnabled}
        answer_id={answerId}
        questionNumber={questionNumber}
        onChange={onAnswerChange}
        feedback_html={feedback_html}
        correct_answer_feedback_html={correct_answer_feedback_html}
        correct_answer_id={is_completed ? correct_answer_id : null}
        incorrectAnswerId={incorrectAnswerId}
        className="step-card-body"
        hideAnswers={false}
        displayFormats={false}
    displaySolution={displaySolution}
      >
        <span>FreeResponseReview @TODO</span>
      </Question>
      <StepCardFooter>
        <div className="points">
          <strong>Points: {available_points}</strong>
          <span className="attempts-left">
            {hasMultipleAttempts &&
              attempts_remaining > 0 &&
              <AttemptsRemaining count={attempts_remaining} />}
          </span>
          <PublishedComments published_comments={published_comments} />
          {detailedSolution && (<div><strong>Detailed solution:</strong> <Content html={detailedSolution} /></div>)}
        </div>
        <div className="controls">
          {canAnswer && needsSaved ?
            <SaveButton
             disabled={apiIsPending || !answerId}
             isWaiting={apiIsPending}
             attempt_number={attempt_number}
             onAnswerSave={onAnswerSave}
            /> :
            <NextButton onNextStep={onNextStep} canUpdateCurrentStep={canUpdateCurrentStep} />}
        </div>
      </StepCardFooter>
    </div>
  );
}
