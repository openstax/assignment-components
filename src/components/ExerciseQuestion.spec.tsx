import { ExerciseQuestion, ExerciseQuestionProps } from './ExerciseQuestion';
import renderer from 'react-test-renderer';

describe('ExerciseQuestion', () => {
  let props: ExerciseQuestionProps;

  beforeEach(() => {
    props = {
      question: {
        id: '1',
        stem_html: 'Is this a question?',
        collaborator_solutions: [],
        formats: [],
        stimulus_html: '',
        answers: [{
          id: '1',
          correctness: undefined,
          content_html: 'True',
        }, {
          id: '2',
          correctness: undefined,
          content_html: 'False',
        }],
        is_answer_order_important: false,
      },
      questionNumber: 1,
      choicesEnabled: false,
      hasMultipleAttempts: false,
      onAnswerChange: () => null,
      onAnswerSave: () => null,
      onNextStep: () => null,
      feedback_html: '',
      correct_answer_feedback_html: '',
      is_completed: false,
      correct_answer_id: '',
      incorrectAnswerId: '',
      multiPartGroup: [],
      answerId: '',
      attempts_remaining: 2,
      published_comments: '',
      detailedSolution: '',
      canAnswer: false,
      needsSaved: false,
      canUpdateCurrentStep: false,
      attempt_number: 0,
      apiIsPending: false,
      displaySolution: false,
      available_points: '',
      exercise_uid: '',
    }
  });

  it('matches snapshot', () => {
    const tree = renderer.create(
      <ExerciseQuestion {...props} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders all attempts remaining', () => {
    const tree = renderer.create(
      <ExerciseQuestion {...props}
        hasMultipleAttempts={true}
        choicesEnabled={true}
        attempts_remaining={2}
        attempt_number={0}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders some attempts remaining', () => {
    const tree = renderer.create(
      <ExerciseQuestion {...props}
        hasMultipleAttempts={true}
        choicesEnabled={true}
        attempts_remaining={1}
        attempt_number={1}
        incorrectAnswerId='2'
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders no attempts remaining', () => {
    const tree = renderer.create(
      <ExerciseQuestion {...props}
        hasMultipleAttempts={true}
        choicesEnabled={false}
        attempts_remaining={0}
        attempt_number={2}
        incorrectAnswerId='2'
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Save button', () => {
    const tree = renderer.create(
      <ExerciseQuestion {...props}
        choicesEnabled={true}
        incorrectAnswerId='2'
        canAnswer={true}
        needsSaved={true}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Re-submit button', () => {
    const tree = renderer.create(
      <ExerciseQuestion {...props}
        choicesEnabled={true}
        incorrectAnswerId='2'
        canAnswer={true}
        needsSaved={true}
        attempt_number={1}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders continue button (unused?)', () => {
    const tree = renderer.create(
      <ExerciseQuestion {...props}
        choicesEnabled={false}
        incorrectAnswerId='2'
        canAnswer={false}
        canUpdateCurrentStep={true}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders detailed solution and published comments', () => {
    const tree = renderer.create(
      <ExerciseQuestion {...props}
        choicesEnabled={false}
        incorrectAnswerId='2'
        correct_answer_id='1'
        is_completed={true}
        canAnswer={false}
        needsSaved={false}
        detailedSolution='A detailed solution'
        published_comments='Teacher feedback'
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
