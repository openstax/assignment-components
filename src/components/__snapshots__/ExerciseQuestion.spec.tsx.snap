// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExerciseQuestion matches snapshot 1`] = `
<div
  data-test-id="student-exercise-question"
>
  <div
    className="sc-bczRLJ bKRYCW openstax-question step-card-body"
    data-question-number={1}
    data-test-id="question"
  >
    <div
      className="question-stem"
      dangerouslySetInnerHTML={
        Object {
          "__html": "Is this a question?",
        }
      }
      data-question-number={1}
    />
    <span>
      FreeResponseReview @TODO
    </span>
    <div
      className="answers-table"
    >
      <div
        className="openstax-answer"
      >
        <section
          className="answers-answer disabled"
          role="region"
        >
          <input
            checked={false}
            className="answer-input-box"
            disabled={true}
            id="1-option-0"
            name="1-options"
            onChange={[Function]}
            type="radio"
          />
          <label
            className="answer-label"
            htmlFor="1-option-0"
          >
            <span
              className="answer-letter-wrapper"
            >
              <button
                aria-label="Choice a:"
                className="answer-letter"
                data-test-id="answer-choice-a"
                disabled={true}
                onClick={[Function]}
              >
                a
              </button>
            </span>
            <div
              className="answer-answer"
            >
              <span
                className="answer-content"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "True",
                  }
                }
              />
            </div>
          </label>
        </section>
      </div>
      <div
        className="openstax-answer"
      >
        <section
          className="answers-answer disabled"
          role="region"
        >
          <input
            checked={false}
            className="answer-input-box"
            disabled={true}
            id="1-option-1"
            name="1-options"
            onChange={[Function]}
            type="radio"
          />
          <label
            className="answer-label"
            htmlFor="1-option-1"
          >
            <span
              className="answer-letter-wrapper"
            >
              <button
                aria-label="Choice b:"
                className="answer-letter"
                data-test-id="answer-choice-b"
                disabled={true}
                onClick={[Function]}
              >
                b
              </button>
            </span>
            <div
              className="answer-answer"
            >
              <span
                className="answer-content"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "False",
                  }
                }
              />
            </div>
          </label>
        </section>
      </div>
    </div>
  </div>
  <div
    className="sc-gsnTZi cfqehf"
  >
    <div
      className="points"
    >
      <strong>
        Points: 
      </strong>
      <span
        className="attempts-left"
      />
    </div>
    <div
      className="controls"
    >
      <button
        className="sc-dkzDqf fVVaSh"
        data-test-id="continue-btn"
        onClick={[Function]}
      >
        Next
      </button>
    </div>
  </div>
</div>
`;

exports[`ExerciseQuestion renders Re-submit button 1`] = `
<div
  data-test-id="student-exercise-question"
>
  <div
    className="sc-bczRLJ bKRYCW openstax-question step-card-body has-incorrect-answer"
    data-question-number={1}
    data-test-id="question"
  >
    <div
      className="question-stem"
      dangerouslySetInnerHTML={
        Object {
          "__html": "Is this a question?",
        }
      }
      data-question-number={1}
    />
    <span>
      FreeResponseReview @TODO
    </span>
    <div
      className="answers-table"
    >
      <div
        className="openstax-answer"
      >
        <section
          className="answers-answer"
          role="region"
        >
          <input
            checked={false}
            className="answer-input-box"
            disabled={false}
            id="1-option-0"
            name="1-options"
            onChange={[Function]}
            type="radio"
          />
          <label
            className="answer-label"
            htmlFor="1-option-0"
          >
            <span
              className="answer-letter-wrapper"
            >
              <button
                aria-label="Choice a:"
                className="answer-letter"
                data-test-id="answer-choice-a"
                disabled={false}
                onClick={[Function]}
              >
                a
              </button>
            </span>
            <div
              className="answer-answer"
            >
              <span
                className="answer-content"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "True",
                  }
                }
              />
            </div>
          </label>
        </section>
      </div>
      <div
        className="openstax-answer"
      >
        <section
          className="answers-answer answer-incorrect"
          role="region"
        >
          <input
            checked={false}
            className="answer-input-box"
            disabled={false}
            id="1-option-1"
            name="1-options"
            onChange={[Function]}
            type="radio"
          />
          <label
            className="answer-label"
            htmlFor="1-option-1"
          >
            <span
              className="answer-letter-wrapper"
            >
              <button
                aria-label="Choice b:"
                className="answer-letter"
                data-test-id="answer-choice-b"
                disabled={true}
                onClick={[Function]}
              >
                b
              </button>
            </span>
            <div
              className="answer-answer"
            >
              <span
                className="answer-content"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "False",
                  }
                }
              />
            </div>
          </label>
        </section>
      </div>
    </div>
  </div>
  <div
    className="sc-gsnTZi cfqehf"
  >
    <div
      className="points"
    >
      <strong>
        Points: 
      </strong>
      <span
        className="attempts-left"
      />
    </div>
    <div
      className="controls"
    >
      <button
        className="sc-dkzDqf fVVaSh"
      >
        Re-submit
      </button>
    </div>
  </div>
</div>
`;

exports[`ExerciseQuestion renders Save button 1`] = `
<div
  data-test-id="student-exercise-question"
>
  <div
    className="sc-bczRLJ bKRYCW openstax-question step-card-body has-incorrect-answer"
    data-question-number={1}
    data-test-id="question"
  >
    <div
      className="question-stem"
      dangerouslySetInnerHTML={
        Object {
          "__html": "Is this a question?",
        }
      }
      data-question-number={1}
    />
    <span>
      FreeResponseReview @TODO
    </span>
    <div
      className="answers-table"
    >
      <div
        className="openstax-answer"
      >
        <section
          className="answers-answer"
          role="region"
        >
          <input
            checked={false}
            className="answer-input-box"
            disabled={false}
            id="1-option-0"
            name="1-options"
            onChange={[Function]}
            type="radio"
          />
          <label
            className="answer-label"
            htmlFor="1-option-0"
          >
            <span
              className="answer-letter-wrapper"
            >
              <button
                aria-label="Choice a:"
                className="answer-letter"
                data-test-id="answer-choice-a"
                disabled={false}
                onClick={[Function]}
              >
                a
              </button>
            </span>
            <div
              className="answer-answer"
            >
              <span
                className="answer-content"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "True",
                  }
                }
              />
            </div>
          </label>
        </section>
      </div>
      <div
        className="openstax-answer"
      >
        <section
          className="answers-answer answer-incorrect"
          role="region"
        >
          <input
            checked={false}
            className="answer-input-box"
            disabled={false}
            id="1-option-1"
            name="1-options"
            onChange={[Function]}
            type="radio"
          />
          <label
            className="answer-label"
            htmlFor="1-option-1"
          >
            <span
              className="answer-letter-wrapper"
            >
              <button
                aria-label="Choice b:"
                className="answer-letter"
                data-test-id="answer-choice-b"
                disabled={true}
                onClick={[Function]}
              >
                b
              </button>
            </span>
            <div
              className="answer-answer"
            >
              <span
                className="answer-content"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "False",
                  }
                }
              />
            </div>
          </label>
        </section>
      </div>
    </div>
  </div>
  <div
    className="sc-gsnTZi cfqehf"
  >
    <div
      className="points"
    >
      <strong>
        Points: 
      </strong>
      <span
        className="attempts-left"
      />
    </div>
    <div
      className="controls"
    >
      <button
        className="sc-dkzDqf fVVaSh"
      >
        Submit
      </button>
    </div>
  </div>
</div>
`;

exports[`ExerciseQuestion renders all attempts remaining 1`] = `
<div
  data-test-id="student-exercise-question"
>
  <div
    className="sc-bczRLJ bKRYCW openstax-question step-card-body"
    data-question-number={1}
    data-test-id="question"
  >
    <div
      className="question-stem"
      dangerouslySetInnerHTML={
        Object {
          "__html": "Is this a question?",
        }
      }
      data-question-number={1}
    />
    <span>
      FreeResponseReview @TODO
    </span>
    <div
      className="answers-table"
    >
      <div
        className="openstax-answer"
      >
        <section
          className="answers-answer"
          role="region"
        >
          <input
            checked={false}
            className="answer-input-box"
            disabled={false}
            id="1-option-0"
            name="1-options"
            onChange={[Function]}
            type="radio"
          />
          <label
            className="answer-label"
            htmlFor="1-option-0"
          >
            <span
              className="answer-letter-wrapper"
            >
              <button
                aria-label="Choice a:"
                className="answer-letter"
                data-test-id="answer-choice-a"
                disabled={false}
                onClick={[Function]}
              >
                a
              </button>
            </span>
            <div
              className="answer-answer"
            >
              <span
                className="answer-content"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "True",
                  }
                }
              />
            </div>
          </label>
        </section>
      </div>
      <div
        className="openstax-answer"
      >
        <section
          className="answers-answer"
          role="region"
        >
          <input
            checked={false}
            className="answer-input-box"
            disabled={false}
            id="1-option-1"
            name="1-options"
            onChange={[Function]}
            type="radio"
          />
          <label
            className="answer-label"
            htmlFor="1-option-1"
          >
            <span
              className="answer-letter-wrapper"
            >
              <button
                aria-label="Choice b:"
                className="answer-letter"
                data-test-id="answer-choice-b"
                disabled={false}
                onClick={[Function]}
              >
                b
              </button>
            </span>
            <div
              className="answer-answer"
            >
              <span
                className="answer-content"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "False",
                  }
                }
              />
            </div>
          </label>
        </section>
      </div>
    </div>
  </div>
  <div
    className="sc-gsnTZi cfqehf"
  >
    <div
      className="points"
    >
      <strong>
        Points: 
      </strong>
      <span
        className="attempts-left"
      >
        <div>
          2
           attempt
          s
           left
        </div>
      </span>
    </div>
    <div
      className="controls"
    >
      <button
        className="sc-dkzDqf fVVaSh"
        data-test-id="continue-btn"
        onClick={[Function]}
      >
        Next
      </button>
    </div>
  </div>
</div>
`;

exports[`ExerciseQuestion renders continue button (unused?) 1`] = `
<div
  data-test-id="student-exercise-question"
>
  <div
    className="sc-bczRLJ bKRYCW openstax-question step-card-body has-incorrect-answer"
    data-question-number={1}
    data-test-id="question"
  >
    <div
      className="question-stem"
      dangerouslySetInnerHTML={
        Object {
          "__html": "Is this a question?",
        }
      }
      data-question-number={1}
    />
    <span>
      FreeResponseReview @TODO
    </span>
    <div
      className="answers-table"
    >
      <div
        className="openstax-answer"
      >
        <section
          className="answers-answer disabled"
          role="region"
        >
          <input
            checked={false}
            className="answer-input-box"
            disabled={true}
            id="1-option-0"
            name="1-options"
            onChange={[Function]}
            type="radio"
          />
          <label
            className="answer-label"
            htmlFor="1-option-0"
          >
            <span
              className="answer-letter-wrapper"
            >
              <button
                aria-label="Choice a:"
                className="answer-letter"
                data-test-id="answer-choice-a"
                disabled={true}
                onClick={[Function]}
              >
                a
              </button>
            </span>
            <div
              className="answer-answer"
            >
              <span
                className="answer-content"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "True",
                  }
                }
              />
            </div>
          </label>
        </section>
      </div>
      <div
        className="openstax-answer"
      >
        <section
          className="answers-answer disabled answer-incorrect"
          role="region"
        >
          <input
            checked={false}
            className="answer-input-box"
            disabled={true}
            id="1-option-1"
            name="1-options"
            onChange={[Function]}
            type="radio"
          />
          <label
            className="answer-label"
            htmlFor="1-option-1"
          >
            <span
              className="answer-letter-wrapper"
            >
              <button
                aria-label="Choice b:"
                className="answer-letter"
                data-test-id="answer-choice-b"
                disabled={true}
                onClick={[Function]}
              >
                b
              </button>
            </span>
            <div
              className="answer-answer"
            >
              <span
                className="answer-content"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "False",
                  }
                }
              />
            </div>
          </label>
        </section>
      </div>
    </div>
  </div>
  <div
    className="sc-gsnTZi cfqehf"
  >
    <div
      className="points"
    >
      <strong>
        Points: 
      </strong>
      <span
        className="attempts-left"
      />
    </div>
    <div
      className="controls"
    >
      <button
        className="sc-dkzDqf fVVaSh"
        data-test-id="continue-btn"
        onClick={[Function]}
      >
        Continue
      </button>
    </div>
  </div>
</div>
`;

exports[`ExerciseQuestion renders detailed solution and published comments 1`] = `
<div
  data-test-id="student-exercise-question"
>
  <div
    className="sc-bczRLJ bKRYCW openstax-question step-card-body has-correct-answer has-incorrect-answer"
    data-question-number={1}
    data-test-id="question"
  >
    <div
      className="question-stem"
      dangerouslySetInnerHTML={
        Object {
          "__html": "Is this a question?",
        }
      }
      data-question-number={1}
    />
    <span>
      FreeResponseReview @TODO
    </span>
    <div
      className="answers-table"
    >
      <div
        className="openstax-answer"
      >
        <section
          className="answers-answer disabled answer-correct"
          role="region"
        >
          <label
            className="answer-label"
            htmlFor="1-option-0"
          >
            <span
              className="answer-letter-wrapper"
            >
              <button
                aria-label="Choice a(Correct Answer):"
                className="answer-letter"
                data-test-id="answer-choice-a"
                disabled={true}
              >
                a
              </button>
            </span>
            <div
              className="answer-answer"
            >
              <span
                className="answer-content"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "True",
                  }
                }
              />
            </div>
          </label>
        </section>
      </div>
      <div
        className="openstax-answer"
      >
        <section
          className="answers-answer disabled answer-incorrect"
          role="region"
        >
          <label
            className="answer-label"
            htmlFor="1-option-1"
          >
            <span
              className="answer-letter-wrapper"
            >
              <button
                aria-label="Choice b(Incorrect Answer):"
                className="answer-letter"
                data-test-id="answer-choice-b"
                disabled={true}
              >
                b
              </button>
            </span>
            <div
              className="answer-answer"
            >
              <span
                className="answer-content"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "False",
                  }
                }
              />
            </div>
          </label>
        </section>
      </div>
    </div>
  </div>
  <div
    className="sc-gsnTZi cfqehf"
  >
    <div
      className="points"
    >
      <strong>
        Points: 
      </strong>
      <span
        className="attempts-left"
      />
      <div>
        <strong>
          Feedback:
        </strong>
         
        Teacher feedback
      </div>
      <div>
        <strong>
          Detailed solution:
        </strong>
         
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "A detailed solution",
            }
          }
        />
      </div>
    </div>
    <div
      className="controls"
    >
      <button
        className="sc-dkzDqf fVVaSh"
        data-test-id="continue-btn"
        onClick={[Function]}
      >
        Next
      </button>
    </div>
  </div>
</div>
`;

exports[`ExerciseQuestion renders no attempts remaining 1`] = `
<div
  data-test-id="student-exercise-question"
>
  <div
    className="sc-bczRLJ bKRYCW openstax-question step-card-body has-incorrect-answer"
    data-question-number={1}
    data-test-id="question"
  >
    <div
      className="question-stem"
      dangerouslySetInnerHTML={
        Object {
          "__html": "Is this a question?",
        }
      }
      data-question-number={1}
    />
    <span>
      FreeResponseReview @TODO
    </span>
    <div
      className="answers-table"
    >
      <div
        className="openstax-answer"
      >
        <section
          className="answers-answer disabled"
          role="region"
        >
          <input
            checked={false}
            className="answer-input-box"
            disabled={true}
            id="1-option-0"
            name="1-options"
            onChange={[Function]}
            type="radio"
          />
          <label
            className="answer-label"
            htmlFor="1-option-0"
          >
            <span
              className="answer-letter-wrapper"
            >
              <button
                aria-label="Choice a:"
                className="answer-letter"
                data-test-id="answer-choice-a"
                disabled={true}
                onClick={[Function]}
              >
                a
              </button>
            </span>
            <div
              className="answer-answer"
            >
              <span
                className="answer-content"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "True",
                  }
                }
              />
            </div>
          </label>
        </section>
      </div>
      <div
        className="openstax-answer"
      >
        <section
          className="answers-answer disabled answer-incorrect"
          role="region"
        >
          <input
            checked={false}
            className="answer-input-box"
            disabled={true}
            id="1-option-1"
            name="1-options"
            onChange={[Function]}
            type="radio"
          />
          <label
            className="answer-label"
            htmlFor="1-option-1"
          >
            <span
              className="answer-letter-wrapper"
            >
              <button
                aria-label="Choice b:"
                className="answer-letter"
                data-test-id="answer-choice-b"
                disabled={true}
                onClick={[Function]}
              >
                b
              </button>
            </span>
            <div
              className="answer-answer"
            >
              <span
                className="answer-content"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "False",
                  }
                }
              />
            </div>
          </label>
        </section>
      </div>
    </div>
  </div>
  <div
    className="sc-gsnTZi cfqehf"
  >
    <div
      className="points"
    >
      <strong>
        Points: 
      </strong>
      <span
        className="attempts-left"
      />
    </div>
    <div
      className="controls"
    >
      <button
        className="sc-dkzDqf fVVaSh"
        data-test-id="continue-btn"
        onClick={[Function]}
      >
        Next
      </button>
    </div>
  </div>
</div>
`;

exports[`ExerciseQuestion renders some attempts remaining 1`] = `
<div
  data-test-id="student-exercise-question"
>
  <div
    className="sc-bczRLJ bKRYCW openstax-question step-card-body has-incorrect-answer"
    data-question-number={1}
    data-test-id="question"
  >
    <div
      className="question-stem"
      dangerouslySetInnerHTML={
        Object {
          "__html": "Is this a question?",
        }
      }
      data-question-number={1}
    />
    <span>
      FreeResponseReview @TODO
    </span>
    <div
      className="answers-table"
    >
      <div
        className="openstax-answer"
      >
        <section
          className="answers-answer"
          role="region"
        >
          <input
            checked={false}
            className="answer-input-box"
            disabled={false}
            id="1-option-0"
            name="1-options"
            onChange={[Function]}
            type="radio"
          />
          <label
            className="answer-label"
            htmlFor="1-option-0"
          >
            <span
              className="answer-letter-wrapper"
            >
              <button
                aria-label="Choice a:"
                className="answer-letter"
                data-test-id="answer-choice-a"
                disabled={false}
                onClick={[Function]}
              >
                a
              </button>
            </span>
            <div
              className="answer-answer"
            >
              <span
                className="answer-content"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "True",
                  }
                }
              />
            </div>
          </label>
        </section>
      </div>
      <div
        className="openstax-answer"
      >
        <section
          className="answers-answer answer-incorrect"
          role="region"
        >
          <input
            checked={false}
            className="answer-input-box"
            disabled={false}
            id="1-option-1"
            name="1-options"
            onChange={[Function]}
            type="radio"
          />
          <label
            className="answer-label"
            htmlFor="1-option-1"
          >
            <span
              className="answer-letter-wrapper"
            >
              <button
                aria-label="Choice b:"
                className="answer-letter"
                data-test-id="answer-choice-b"
                disabled={true}
                onClick={[Function]}
              >
                b
              </button>
            </span>
            <div
              className="answer-answer"
            >
              <span
                className="answer-content"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "False",
                  }
                }
              />
            </div>
          </label>
        </section>
      </div>
    </div>
  </div>
  <div
    className="sc-gsnTZi cfqehf"
  >
    <div
      className="points"
    >
      <strong>
        Points: 
      </strong>
      <span
        className="attempts-left"
      >
        <div>
          1
           attempt
           left
        </div>
      </span>
    </div>
    <div
      className="controls"
    >
      <button
        className="sc-dkzDqf fVVaSh"
        data-test-id="continue-btn"
        onClick={[Function]}
      >
        Next
      </button>
    </div>
  </div>
</div>
`;
